src/client/opamCommands.ml

```diff
@@ -2549,12 +2549,13 @@ let switch cli =
          $(b,--no-install) is specified."
         OpamArg.dir_sep OpamSwitch.external_dirname);
     `P (Printf.sprintf
-         "$(b,opam switch set) sets the default switch globally, but it is also \
-         possible to select a switch in a given shell session, using the \
-         environment. For that, use $(i,%s)."
-        OpamEnv.(
-          shell_eval_invocation shell
-            (opam_env_invocation ~switch:"SWITCH" ~set_opamswitch:true shell)
+          "$(b,opam switch set) sets the default switch globally. If shell \
+           hook is enabled, current shell session is always synchronised with \
+           selected switch (or automatically selected local switch). \
+           Otherwise, you can use $(i,%s)."
```

Comment:
Something like that to be more explicit (not sure how to turn the phrase)
```suggestion
           Otherwise, you can use $(i,%s) to force the current switch to be the one set (overrides local switches)."
```